### DISCUSSION TITLE
boost::geometry::sym_difference -- is it expected to return a polygon with spikes?

### DISCUSSION THREAD

**Discussion by vschoech** [boostorg/geometry#929] [Q&A] [answered]
In some instances, `boost::geometry::sym_difference` seems to return a polygon with spikes. Is this expected behavior?

I'm trying to decide whether I should make my code deal with those spikes, or file a bug report. I assume it's the former, but it seems to occur rather rarely so I thought I'd rather ask for your opinion on this. The documentation isn't explicit in this regard:
https://www.boost.org/doc/libs/1_77_0/libs/geometry/doc/html/geometry/reference/algorithms/sym_difference/sym_difference_3.html

Thanks!
Volker

**Comment by awulkiew** [comment, marked as answer]
If the input of an algorithm is valid then the output should be valid as well. There should be no spikes. Unless these are just visually spikey parts of a geometry but the geometry is valid.
